File: php/features.php
<?php
declare(strict_types=1);

// Ipleak SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IpleakFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IpleakBaseFeature();
            case "test":
                return new IpleakTestFeature();
            default:
                return new IpleakBaseFeature();
        }
    }
}

File: py/features.py
# Ipleak SDK feature factory

from feature.base_feature import IpleakBaseFeature
from feature.test_feature import IpleakTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IpleakBaseFeature(),
        "test": lambda: IpleakTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
